bundles/amazon/ack/elasticache/bundle/config/01-replication-group.ytt.yml

```diff
@@ -0,0 +1,225 @@
+#@ load("@ytt:data", "data")
+#@ load("@ytt:assert", "assert")
+
+#@ def get_id(x): return "{}-{}".format(data.values.name, x)
+
+#@ def tags(resource_tags=[]):
+#@   return [ x for x in (data.values.tags + resource_tags) if x["key"] != "" and x["value"] != "" ]
```

Comment:
**question** Kind of unfortunate we have to special case `key` and `value here. Do we need to set these in the schema?

Reply:
Sorry, I didn't get the question. What do you mean?

Reply:
Maybe asked another way, why do we need this statement?

```code
if x["key"] != "" and x["value"] != ""
```

Reply:
It's for validating the inputs.
Without that statement the following input
```yaml
tags:
  - value: a value without a key
```
would produce
```yaml
apiVersion: elasticache.services.k8s.aws/v1alpha1
kind: ReplicationGroup
metadata:
  name: trp01
  namespace: elasticache-01
spec:
  description: A redis service instance
  engine: redis
  engineVersion: "6.2"
  replicationGroupID: trp01
  cacheNodeType: cache.t2.micro
  cacheSubnetGroupName: trp
  cacheParameterGroupName: trp01
  securityGroupIDs:
  - sg-0a4ddae4fbf426cc8
  atRestEncryptionEnabled: true
  transitEncryptionEnabled: true
  replicasPerNodeGroup: 0
  userGroupIDs:
  - trp01
  tags:
  - value: a value without a key
    key: ""
```

Reply:
Understood, makes sense 👍

Reply:
Until we have proper input validation built into ytt we have to use workarounds like this..